kernel/src/hil/qdec.rs

```diff
@@ -0,0 +1,33 @@
+//! Interface for a Qdec compatible chip
+//!
+//! This trait provides a stanfard interface for chips with a
+//! quadrature decoder. Note this interface is experimental and
+//! may need further updates once implemented on additional chips
+
+use crate::returncode::ReturnCode;
+
+pub trait QdecDriver {
+    /// Sets the client which will receive interrupts
+    fn set_client(&self, client: &'static dyn QdecClient);
+
+    /// Enables the SAMPLERDY interrupt
+    fn enable_interrupts(&self) -> ReturnCode;
+
+    /// Enables the Qdec, returning error if Qdec does not exist
+    fn enable_qdec(&self) -> ReturnCode;
+
+    /// Checks if the qdec has been enabled
+    fn enabled(&self) -> ReturnCode;
+
+    /// Reads the accumulator value and resets it
+    /// Note accumulator means the measure of how many ticks the
+    /// QDEC has moved since the last time the function was called
+    fn get_acc(&self) -> u32;
```

Comment:
I'm surprised that this is a `u32`, how is direction encoded?

Reply:
Thanks for the comment! The main motivation for using u32 was the eventual conversion to usize that is done in the capsule because the value passed into the callback schedule function needs to be a usize to my knowledge (When attempting this conversion with an i32, it presented some challenges). Since the value actually provided to userspace is an int, the cast correctly changes the overflow from turned the rotary encoder backwards into negative values. Thus, given the testing done so far, it seems the direction is accurately calculated and communicated to the user. However, if you have further input on how to make this process more bulletproof and reliable then I'd be happy to discuss further!

Reply:
Upon further consideration, the u32 has now been changed to an i32 to better represent the position value!